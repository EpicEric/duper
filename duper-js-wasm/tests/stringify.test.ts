import { describe, expect, it } from "vitest";
import { stringify } from "..";

describe("stringify", () => {
  it("stringifies a basic object", () => {
    const data = {
      name: "Wireless Headphones",
      price: 129.99,
      in_stock: true,
      tags: ["electronics", "audio"],
    };
    const duper_string = stringify(data);
    expect(duper_string).toMatchSnapshot();
  });

  it("stringifies nested structures", () => {
    const data = ["foo", { bar: ["baz", { qux: null }] }];
    const duper_string = stringify(data);
    expect(duper_string).toMatchSnapshot();
  });
});
