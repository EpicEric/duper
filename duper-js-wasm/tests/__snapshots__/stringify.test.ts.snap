// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`stringify > stringifies a basic object 1`] = `"{name: "Wireless Headphones", price: 129.99, in_stock: true, tags: ["electronics", "audio"]}"`;

exports[`stringify > stringifies nested structures 1`] = `"["foo", {bar: ["baz", {qux: null}]}]"`;
